pandas/_libs/tslibs/tzconversion.pyx

```diff
@@ -60,8 +61,8 @@ cdef int64_t tz_localize_to_utc_single(
     elif is_utc(tz) or tz is None:
         return val
 
-    elif is_tzlocal(tz):
-        return _tz_convert_tzlocal_utc(val, tz, to_utc=True)
+    elif is_tzlocal(tz) or is_zoneinfo(tz):
+        return _tz_localize_using_tzinfo_api(val, tz, to_utc=True)
```

Comment:
I think this should be the default behavior, rather than something triggered only for zoneinfo objects.

Reply:
im coming around to a similar opinion.  Probably for a separate PR since that would involve a lot of code shuffling (attempts to unify the many places we do these checks have been stymied xref #46397, #46246)